pico/api/src/main/java/io/helidon/pico/spi/CallingContextCreator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.pico.spi;
+
+import java.util.Optional;
+
+import io.helidon.pico.PicoServices;
+
+/**
+ * Factory for creating {@link CallingContext} and builders for the calling context.
+ * After a calling context builder is created, it should be amended with as much contextual information as possible, and then
+ * optionally set globally using {@link CallingContext#globalCallingContext(CallingContext, boolean)}.
+ */
+public class CallingContextCreator {
```

Comment:
A new public class?
The name should maybe be a Factory?
Is this something we expect our customers to use? If not, this class should not exist as public

Reply:
I would have preferred to use CallingContext, but it needs to forward reference the builders that will code generate.  And this will create a compiler error. The class needs to exist as public since it is called from maven-plugin, etc.

Reply:
pushed